roles/kubernetes/preinstall/tasks/0081-ntp-configurations.yml

```diff
@@ -0,0 +1,65 @@
+---
+- name: Ensure NTP package
+  package:
+    name:
+      - "{{ ntp_package }}"
+    state: present
+
+- name: Disable systemd-timesyncd
+  service:
+    name: systemd-timesyncd.service
+    enabled: false
+    state: stopped
+  failed_when: false
+
+- name: Set fact NTP settings
+  set_fact:
+    ntp_config_file: >-
+      {% if ntp_package == "ntp" -%}
+      /etc/ntp.conf
+      {%- elif ansible_os_family == 'RedHat' -%}
```

Comment:
If `ntp_package` is not `ntp`, `ansible_os_family` is always `RedHat` because of line 72-77 on roles/kubernetes/preinstall/defaults/main.yml.
Which os do we need to take care of here?

Reply:
Hi @oomichi , There are a special case.

The user specify the ` ntp_package=chrony ` in Ubuntu,  because the user want to use chrony in Ubuntu

The default chrony file path in Ubuntu/Debian is `/etc/chrony/chrony.conf` . Refer to (https://ubuntu.com/blog/ubuntu-bionic-using-chrony-to-configure-ntp),
And the default chrony file in Centos/Redhat/SUSE is /etc/chrony.conf. Refer to ( https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/ch-configuring_ntp_using_the_chrony_suite, https://documentation.suse.com/sles/15-SP1/html/SLES-all/cha-ntp.html)

Reply:
Thank you for your explanation, I see that.